### PR TITLE
Implement filtering by references

### PR DESCRIPTION
**Pull request by guergana**
first commit: only with references.
second commit: `without` reference case.
third commit: added missing triple.
fourth commit: enable ReferenceRelationDropdown

### CODE REVIEW

File: src/sparql/TripleBuilder.ts [hunk 1]
```diff
@@ -101,6 +103,23 @@ export default class TripleBuilder {
 		return items;
 	}
 
+	public buildTripleForFilter( condition: Condition ): Triple {
```
Comment (Ladsgroup) [inline src/sparql/TripleBuilder.ts]: Looking at the triple, it doesn't look like it's a triple for Filter. It looks like a reference triple ("wasDerivedFrom"). What about naming it "buildReferenceTriple"?

File: src/sparql/TripleBuilder.ts [hunk 2]
```diff
@@ -2,13 +2,15 @@ import { Condition } from '@/sparql/QueryRepresentation';
...
+import ReferenceRelation from '@/data-model/ReferenceRelation';
 
 export default class TripleBuilder {
 	public buildTripleFromQueryCondition( condition: Condition ): Triple {
 		return {
 			subject: {
 				termType: 'Variable',
-				value: 'item',
+				value: condition.referenceRelation === ReferenceRelation.Regardless ?
+					'item' : 'statement' + condition.propertyId,
```
Comment (Ladsgroup) [inline src/sparql/TripleBuilder.ts]: Technically it's correct but we can have a property being repeated more than once (occupation matching artist and occupation matching engineer) so maybe adding condition index would be better?

Comment (guergana) [reply src/sparql/TripleBuilder.ts]: Yes, agree. I mentioned it at the top of the conversation here. That's one of the things that are still missing. Thanks for the suggestion.

File: src/sparql/TripleBuilder.ts (hunk 1 above)
Comment (guergana) [reply src/sparql/TripleBuilder.ts]: I just named it the first thing that came to my mind, yep `buildReferenceTriple` sounds better, will update.

File: src/sparql/QueryObjectBuilder.ts [hunk 3]
```diff
@@ -123,11 +123,19 @@ export default class QueryObjectBuilder {
 			triples: [ this.tripleBuilder.buildTripleFromQueryCondition( condition, conditionIndex ) ],
 		};
 
-		const existsReferenceFilter: FilterPattern = {
+		let referenceExistsOrNot = '';
```
Comment (guergana) [inline src/sparql/QueryObjectBuilder.ts:147]: Not 100% sure about this. If I don't initialize it the `expression` in `referenceFilter` complains. Open to suggestions here.

Comment (Ladsgroup) [reply src/sparql/QueryObjectBuilder.ts:147]: I think it's good TBH

**Review by Ladsgroup — APPROVED** [review]

File: src/sparql/QueryObjectBuilder.ts (hunk 3 above)
Comment (guergana) [reply src/sparql/QueryObjectBuilder.ts:147]: Haha. Can't believe this is approved. (Next comes refactoring)

**Pull request merged**